apps/rush-lib/src/utilities/install-rush-for-ci.ts

```diff
@@ -0,0 +1,173 @@
+// This script is invoked by the CI build, via a build definition step.
+//
+// 'npm install @microsoft/rush -g' will always delete and recreate the rush
+// global folder, even if it is already up to date. This causes a race condition
+// when multiple builds are running simultaneously on the same build machine.
+//
+// As a workaround, this script checks whether Rush is up to date before
+// running the command.
+
+import * as childProcess from 'child_process';
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import { IPackageJson } from '@microsoft/node-core-library';
+
+const PACKAGE_NAME: string = '@microsoft/rush';
+const RUSH_JSON_FILENAME: string = 'rush.json';
+
+let rushJsonDirectory: string = undefined!;
+let basePath: string = __dirname;
+let tempPath: string = __dirname;
+do {
+  const testRushJsonPath: string = path.join(basePath, RUSH_JSON_FILENAME);
+  if (fs.existsSync(testRushJsonPath)) {
+    rushJsonDirectory = basePath;
+    break;
+  } else {
+    basePath = tempPath;
+  }
+} while (basePath !== (tempPath = path.resolve(basePath, '..'))); // Exit the loop when we hit the disk root
+
+if (!rushJsonDirectory) {
+  console.error('Unable to find rush.json.');
+  process.exit(1);
+}
+
+let expectedVersion: string = undefined!;
+const rushJsonPath: string = path.join(rushJsonDirectory, RUSH_JSON_FILENAME);
+try {
+  const rushJsonContents: string = fs.readFileSync(rushJsonPath, 'UTF-8');
+  // Use a regular expression to parse out the rushVersion value because rush.json supports comments,
+  // but JSON.parse does not and we don't want to pull in more dependencies than we need to in this script.
+  const rushJsonMatches: string[] = rushJsonContents.match(/\"rushVersion\"\s*\:\s*\"([0-9a-zA-Z.+\-]+)\"/)!;
+  expectedVersion = rushJsonMatches[1];
+} catch (e) {
+  console.error(
+    `Unable to determine the required version of Rush from rush.json (${rushJsonDirectory}). ` +
+    'The \'rushVersion\' field is either not assigned in rush.json or was specified ' +
+    'using an unexpected syntax.'
+  );
+  process.exit(1);
+}
+
+let npmPath: string = undefined!;
+try {
+  if (os.platform() === 'win32') {
+    // We're on Windows
+    const whereOutput: string = childProcess.execSync('where npm', { stdio: [] }).toString();
+    const lines: string[] = whereOutput.split(os.EOL).filter((line) => !!line);
+    npmPath = lines[lines.length - 1];
+  } else {
+    // We aren't on Windows - assume we're on *NIX or Darwin
+    npmPath = childProcess.execSync('which npm', { stdio: [] }).toString();
+  }
+} catch (e) {
+  console.error(`Unable to determine the path to the NPM tool: ${e}`);
+  process.exit(1);
+}
+
+npmPath = npmPath.trim();
+console.log(os.EOL + `NPM executable is '${npmPath}'`);
+
+if (!fs.existsSync(npmPath)) {
+  console.error('The NPM executable does not exist');
+  process.exit(1);
+}
+
+const rushPathParts: string[] = ['common', 'temp', 'local-rush'];
```

Comment:
>local-rush [](start = 52, length = 10)

Let's rename this to "ci-rush".

Otherwise there will be a weird incongruity with "common/temp/pnpm-local" being backwards from "local-rush"